Documentation/Functions/_StdWrap/_UserFunc.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyVendor\MySitePackage\UserFunctions;
+
+use Psr\Http\Message\ServerRequestInterface;
+use TYPO3\CMS\Core\LinkHandling\LinkService;
+use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+use TYPO3\CMS\Frontend\Typolink\LinkResult;
+use TYPO3\CMS\Frontend\Typolink\LinkResultInterface;
+
+class TypoLinkUserFunc
+{
+    public function createUserFuncLink(
+        LinkResultInterface $content,
+        array $conf,
+        ServerRequestInterface $request,
+    ): LinkResultInterface {
+
+        // First check what kind of link this is.
+        // This example only operates on TYPE_PAGE for internal
+        // links; the userFunc can of course also react to
+        // types like TYPE_URL (external pages)
+        if ($content->getType() === LinkService::TYPE_PAGE) {
+            $cObj = $this->getContentObjectRenderer($request);
+
+            // Check if typolink.userFunc additional data is set
+            // and act on it
+            if ($conf['eventUid.']['data'] ?? null) {
+                $url = $content->getAttribute('href')
```

Comment:
This may break, because there is no cHash attached to the URI.

Reply:
IMO for the sake of the example that's ok, I wouldn't want to make this any more complex at this point, would that be acceptable for you?